### ISSUE TITLE
Unable to use `revision_traffic` without an `image`

### ISSUE THREAD

**Issue by jacobwgillespie** [google-github-actions/deploy-cloudrun#537]
### TL;DR

Previously it was possible to use this action with a `revision_traffic` as a standalone step, which would call `gcloud run services update-traffic`. After v2.7.0, the standalone step will try to call `gcloud run deploy` instead and will error as it is lacking an `image`.

### Expected behavior

It would be nice to maintain the ability to separate the deploy step and the traffic shift step. If this is however not intended, a note about this change should be added to the CHANGELOG.

### Observed behavior

Using `revision_traffic` by itself now calls `gcloud run deploy` and errors

### Action YAML

```yaml
    steps:
      - name: Deploy to Cloud Run
        uses: google-github-actions/deploy-cloudrun@v2
        with:
          project_id: ID
          region: us-east4
          service: api
          image: us-docker.pkg.dev/path/to/image:${{ github.sha }}
          suffix: ${{ github.sha }}

      - name: Shift traffic to deployment
        uses: google-github-actions/deploy-cloudrun@v2
        with:
          project_id: ID
          region: us-east4
          service: api
          revision_traffic: api-${{ github.sha }}=100
```

### Log output

```text
Error: google-github-actions/deploy-cloudrun failed with: failed to execute gcloud command `gcloud run deploy api --update-labels ^,^managed-by=github-actions,commit-sha=xxxxxxxxxxxxxxxxxxx --format json --region us-east4 --project ID`: ERROR: (gcloud.run.deploy) Missing required argument [--image]: Requires a container image to deploy (e.g. `gcr.io/cloudrun/hello:latest`) if no build source is provided.
```

**Comment by davideme** [comment]
We have the same issue, happy to change the syntax but how can we run the `cloud run services update-traffic` equivalent with the github action since 2.7.0?

**Comment by lfbergee** [comment]
For me removing the second call to deploy-cloudrun. Seems like you can set revision_traffic in first call now.

```
    steps:
      - name: Deploy to Cloud Run
        uses: google-github-actions/deploy-cloudrun@v2
        with:
          project_id: ID
          region: us-east4
          service: api
          image: us-docker.pkg.dev/path/to/image:${{ github.sha }}
          suffix: ${{ github.sha }}
          revision_traffic: api-${{ github.sha }}=100
```

**Comment by mjvankampen** [comment]
Tbh shouldn't this have just been a major version bump? We ran into this as well.

**Comment by templarfelix** [comment]
this
`Potentially breaking changes - this fixes a bug where, when tag_traffic or revision_traffic was specified, no deployments happened. This was a very bad bug, but some users may have been depending on the broken behavior. After this version, deployment always happens, and then if tag_traffic or revision_traffic are set, a second command is run to update the traffic accordingly.`


Don't work with metadata; metadata tag_traffic stopped working for me.

**Comment by sethvargo** [comment]
Refs https://github.com/google-github-actions/deploy-cloudrun/issues/533

Hey folks - I'm sorry this disrupted your workflows. This change and its potential impact was [called out in the CHANGELOG](https://github.com/google-github-actions/deploy-cloudrun/releases/tag/v2.7.0) as big as I could. The reason this was not a major version bump is because the previous behavior was a (very bad) bug, and it was never the intended behavior.

Given the job of this GitHub Action is to deploy to Cloud Run, it was very unexpected (and unintentional) that adding an input skips deployment.

@templarfelix - the reason it no longer works for your knative YAML is because the previous behavior was broken. Previously if you set `revision_traffic` or `tag_traffic`, _no_ deployment would happen, so your Knative YAML was not actually being deployed. Now that it's always being deployed, there's an edge case that I missed. That edge case is fixed in https://github.com/google-github-actions/deploy-cloudrun/pull/538.

**Comment by sethvargo** [comment]
@gervaila @melldson #539

**Comment by thefuga** [comment]
> Refs #533
> 
> Hey folks - I'm sorry this disrupted your workflows. This change and its potential impact was [called out in the CHANGELOG](https://github.com/google-github-actions/deploy-cloudrun/releases/tag/v2.7.0) as big as I could. The reason this was not a major version bump is because the previous behavior was a (very bad) bug, and it was never the intended behavior.
> 
> Given the job of this GitHub Action is to deploy to Cloud Run, it was very unexpected (and unintentional) that adding an input skips deployment.

I understand the concern, but this was the behavior the previous version offered, so those users relying on the bug, were relying on it for a reason: it was the provided behavior.
I don't see how breaking other users' workflows helps in your bug fix. Anyone possibly affected by this bug was probably aware of it, whereas people who were relying on it have a breaking change in a minor bump.

**Comment by sethvargo** [comment]
@gervaila @melldson new version is out that introduces the `update_traffic_flags` and manages flags separately.

**Comment by melldson** [comment]
@sethvargo `update_traffic_flags` has the same behavior of flags? I'm trying to use `update_traffic_flags: "--min-instances=10"` but doesn't work.

**Comment by sethvargo** [comment]
@melldson `--min-instances` is not a flag on `gcloud run services update-traffic`. If you want to set min instances on the deployment, set it in `flags` (not `update_traffic_flags`).

**Comment by melldson** [comment]
@sethvargo Yes, but I can't use `flags` and `tag_traffic` together, so I need to deploy twice to set the flags and another one to ensure the traffic is set on my revision?

**Comment by sethvargo** [comment]
@melldson - you can, that's what I fixed in the PR. `flags` applies to the `deploy` command. `update_traffic_flags` applies to the `update-traffic` command.

**Comment by melldson** [comment]
@sethvargo thanks.

**Comment by sethvargo** [comment]
Hey folks - I'm going to close this out since all the major bugs in the latest release have been addressed. Thanks for all the feedback!
